### PR TITLE
Iss1446

### PR DESCRIPTION
**Pull request by AaronJhaj**
added --result that allows multiple inputs such as --result passed,failed,envfail

### CODE REVIEW

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-cli-pr-115-nf8nc for details. If you are unable to do so, please contact a member of the Galasa team.

File: docs/generated/errors-list.md [hunk 1]
```diff
@@ -85,6 +85,8 @@ The `galasactl` tool can generate the following errors:
 - GAL1082E: Invalid time unit specified '{}' in the '--age' parameter. Age of the test runs should be specified in the format '{FROM}{TIME-UNIT}:{TO}{TIME-UNIT}' or '{FROM}{TIME-UNIT}', where 'FROM' is a positive, non-zero integer, 'TO' is a non-negative integer, and 'TIME-UNIT' can be 'w' (weeks), 'd' (days) or 'h' (hours). 'FROM' must be greater than 'TO'. 'TO' defaults to 0 if not specified.
 - GAL1083E: No artifacts to download for run: '{}'
 - GAL1084E: Communications failure while closing an HTTP response. '{}'
+- GAL1085E: Invalid result parameter {}. The possible result values in RAS are: {}
+- GAL1086E: Could not retrieve ruesult names. Reason: '{}'
```
Comment (techcobweb) [inline docs/generated/errors-list.md:89]: spelling "reusult"

File: docs/generated/galasactl_runs_get.md [hunk 2]
```diff
@@ -18,6 +18,7 @@ galasactl runs get [flags]
   -h, --help               help for get
       --name string        the name of the test run we want information about
       --requestor string   the requestor of the test run we want information about
+      --result string      the result of the test run we want information about
```
Comment (techcobweb) [inline docs/generated/galasactl_runs_get.md:21]: A filter on the test runs we want information about. Optional. Default is to display test runs with any result. Case insensitive. Value can be a single value or a comma-separated list. For example "--result Failed,Ignored,EnvFail"

File: pkg/errors/errorMessage.go [hunk 3]
```diff
@@ -155,6 +155,8 @@ var (
 	GALASA_ERROR_BAD_TIME_UNIT_AGE_SPECIFIED    = NewMessageType("GAL1082E: Invalid time unit specified '%s' in the '--age' parameter. Age of the test runs should be specified in the format '{FROM}{TIME-UNIT}:{TO}{TIME-UNIT}' or '{FROM}{TIME-UNIT}', where 'FROM' is a positive, non-zero integer, 'TO' is a non-negative integer, and 'TIME-UNIT' can be 'w' (weeks), 'd' (days) or 'h' (hours). 'FROM' must be greater than 'TO'. 'TO' defaults to 0 if not specified.", 1082, STACK_TRACE_NOT_WANTED)
 	GALASA_ERROR_NO_ARTIFACTS_TO_DOWNLOAD       = NewMessageType("GAL1083E: No artifacts to download for run: '%s'", 1083, STACK_TRACE_NOT_WANTED)
 	GALASA_ERROR_HTTP_RESPONSE_CLOSE_FAILED     = NewMessageType("GAL1084E: Communications failure while closing an HTTP response. '%s'", 1084, STACK_TRACE_WANTED)
+	GALASA_ERROR_INVALID_RESULT_ARGUMENT        = NewMessageType("GAL1085E: Invalid result parameter %s. The possible result values in RAS are: %s", 1085, STACK_TRACE_NOT_WANTED)
```
Comment (techcobweb) [inline pkg/errors/errorMessage.go:158]: Invalid '--result' pararameter value '%s'. The possible result values currently in the Ecosystem Result Archive Store (RAS) are %s

File: pkg/errors/errorMessage.go [hunk 4]
```diff
@@ -155,6 +155,8 @@ var (
...
+	GALASA_ERROR_QUERY_RESULTNAMES_FAILED       = NewMessageType("GAL1086E: Could not retrieve ruesult names. Reason: '%s'", 1086, STACK_TRACE_WANTED)
```
Comment (techcobweb) [inline pkg/errors/errorMessage.go:159]: Communications problem between the command-line tool and the target Galasa Ecosystem. The tool could not retrieve the list of valid result names. Reason: '%s'

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-cli-pr-115-nf8nc-r-kgggr for details. If you are unable to do so, please contact a member of the Galasa team.

File: pkg/runs/resultParameterChecker.go [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+	rasResultNamesData, httpResponse, err := restClient.ResultArchiveStoreAPIApi.GetRasResultNames(context).Execute()
+
+	if err == nil {
+		if httpResponse.StatusCode != http.StatusOK {
+			httpError := "\nhttp response status code: " + strconv.Itoa(httpResponse.StatusCode)
+			errString := err.Error() + httpError
+			err = errors.NewGalasaError(errors.GALASA_ERROR_QUERY_RESULTNAMES_FAILED, errString)
+		} else {
+			rasResultNames := rasResultNamesData.GetResultnames()
+			resultInputs := strings.Split(resultInputString, ",")
```
Comment (techcobweb) [inline pkg/runs/resultParameterChecker.go:34]: Might be worth logging the list of result names to the log ?

File: pkg/runs/resultParameterChecker.go [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+			// compare the input result to the list of possibles - case insensitive
+			for _, resultInput := range resultInputs {
+				matched := false
+				for _, rasResultName := range rasResultNames {
```
Comment (techcobweb) [inline pkg/runs/resultParameterChecker.go:39]: If you put the valid result names into a map[string], then you could just see if that string was in the map without this linear search. Simpler code ?

File: pkg/runs/resultParameterChecker.go [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+	}
+
+	return resultQuery, err
+}
+
+func covertArrayToCommaSeparatedStringWithQuotes(array []string) string {
+
+	sort.Strings(array)
+
+	outputString := strings.Builder{}
```
Comment (techcobweb) [inline pkg/runs/resultParameterChecker.go:66]: You should probably trim leading/trailing whitespace off the user input ?

File: pkg/runs/runsGet.go [hunk 8]
```diff
@@ -179,6 +184,7 @@ func GetRunDetailsFromRasSearchRuns(runs []galasaapi.Run, apiServerUrl string) (
 func GetRunsFromRestApi(
 	runName string,
 	requestor string,
+	result string,
```
Comment (techcobweb) [inline pkg/runs/runsGet.go:187]: It's odd passing a variable called result into a function ?
Perhaps resultFilter would be clearer ?

File: pkg/runs/runsGet.go [hunk 9]
```diff
@@ -46,6 +46,7 @@ func GetRuns(
 	runName string,
 	age string,
 	requestor string,
+	result string,
```
Comment (techcobweb) [inline pkg/runs/runsGet.go:49]: It's odd passing a variable called result into a function ?
Perhaps resultFilter would be clearer ?

requestor might be better as requestorFilter also ?

**Review by techcobweb — APPROVED** [review]
Some comments, but approved assuming you address them.

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-cli-pr-115-nf8nc-r-hp6hh for details. If you are unable to do so, please contact a member of the Galasa team.

File: pkg/runs/resultParameterChecker.go (hunk 7 above)
Comment (AaronJhaj) [reply pkg/runs/resultParameterChecker.go:66]: i thought this too but the command line would take 'Passed' as the argument here and fail at the comma. so trimming is unnecessary

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-cli-pr-115-bxqpk for details. If you are unable to do so, please contact a member of the Galasa team.

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-cli-pr-115-2p7kw for details. If you are unable to do so, please contact a member of the Galasa team.

**Comment by galasa-team** [comment]
Build successful

**Pull request merged**